packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1091,16 +1103,22 @@ def process_payload(self, payload: BaseTxPayload) -> None:
             )
 
         sender = payload.sender
-        if sender not in self.synchronized_data.participants:
+        if sender not in self.accepting_payloads_from:
             raise ABCIAppInternalError(
-                f"{sender} not in list of participants: {sorted(self.synchronized_data.participants)}"
+                f"{sender} not in list of participants: {sorted(self.accepting_payloads_from)}"
             )
 
         if sender in self.collection:
             raise ABCIAppInternalError(
                 f"sender {sender} has already sent value for round: {self.round_id}"
             )
 
+        if self._hash_length:
```

Comment:
Why name this attribute `_hash_length`? I mean it could be made up of any type of fixed-size items, not just hashes as it happened in the current case with Tx settlement skill.

Reply:
It is only used by the `SynchronizeLateMessagesRound`. I suggest to add support for checks on other payloads during payload redesign, for which there exists a separate ticket.

Reply:
So just to clarify, would this be removed during the redesign?

Reply:
I would say most likely so. This information should be present on the payload class in my oppinion, not on the round.